src/instance.rs

```diff
@@ -321,6 +321,14 @@ impl<'a, 'py> Borrowed<'a, 'py, PyAny> {
     pub(crate) unsafe fn from_ptr_unchecked(py: Python<'py>, ptr: *mut ffi::PyObject) -> Self {
         Self(NonNull::new_unchecked(ptr), PhantomData, py)
     }
+
+    /// Converts this `PyAny` to a concrete Python type without checking validity.
+    ///
+    /// # Safety
+    /// Callers must ensure that the type is valid or risk type confusion.
+    pub(crate) unsafe fn downcast_unchecked<T>(self) -> Borrowed<'a, 'py, T> {
+        Borrowed(self.0, PhantomData, self.2)
+    }
```

Comment:
I found no way to downcast a `Borrowed<PyAny>` to a `Borrowed<T>`, so I added this here. One think to note here is, that this shadows the implementation on `PyAnyMethods`.

Reply:
At some point I remember wanting this too, although I called it `downcast_into_unchecked`. Because of the `Copy` semantic, I think that the `downcast_unchecked` you chose is the better name here 👍 (there's no need to ever have `_into` variant of this method to separate ownership transfer).

As this is crate-private I'm completely happy with having this as-is, and if we have problems with shadowing then we can think of a better name.